iterative/utils/analytics.go

```diff
@@ -155,10 +156,57 @@ func GroupId() (string, error) {
 		return "", err
 	}
 
-	return id.String(), nil
+	return id, nil
+}
+
+func readId(path string) (string, error) {
+	file, err := os.Open(path)
+	if file != nil {
+		return "", err
+	}
+	defer file.Close()
+
+	bytes, err := ioutil.ReadAll(file)
+	if err != nil {
+		return "", err
+	}
+
+	var data map[string]interface{}
+
+	if err := json.Unmarshal([]byte(bytes), &data); err != nil {
+		uid, uidError := uuid.FromBytes(bytes)
+		if uidError != nil {
+			return "", fmt.Errorf("failed parsing user_id as json and plaintext: %w", uidError)
+		}
+		logrus.Traceln(fmt.Errorf("found old format telemtry uid, json err: %w", err))
+		return uid.String(), nil
+	}
+
+	if id, ok := data["user_id"].(string); ok {
+		return id, nil
+	}
+
+	return "", errors.New("user_id not found or not a string")
+}
+
+func writeId(path string, id string) error {
+	if err := os.MkdirAll(filepath.Dir(path), 0644); err != nil {
+		return err
+	}
+
+	data := map[string]string{"user_id": id}
+
+	bytes, err := json.MarshalIndent(data, "", "  ")
+	if err != nil {
+		return err
+	}
+
+	return ioutil.WriteFile(path, bytes, 0644)
 }
 
 func UserId() (string, error) {
+	var id string
+	var err error
```

Comment:
Not sure why do you prefer this 🙂

Reply:
there are multiple uses of that var at different scopes, IIRC there was a listing error that `:=` was already defined or something.